hal/src/stm32f2xx/sleep_hal.cpp

```diff
@@ -111,6 +133,11 @@ static int configGpioWakeupSource(const hal_wakeup_source_base_t* wakeupSources)
             irqConf.keepHandler = 1;
             irqConf.keepPriority = 1;
             HAL_Interrupts_Attach(gpioWakeup->pin, nullptr, nullptr, gpioWakeup->mode, &irqConf);
+            
+            Hal_Pin_Info* pinMap = HAL_Pin_Map();
+            uint8_t pinSource = pinMap[gpioWakeup->pin].gpio_pin_source;
+            extiPriority[pinSource] = NVIC_GetPriority(static_cast<IRQn_Type>(GPIO_IRQn[pinSource]));
+            NVIC_SetPriority(static_cast<IRQn_Type>(GPIO_IRQn[pinSource]), 1);
```

Comment:
Nit: This can be done when calling `HAL_Interrupts_Attach()` instead in `IRQChannelPreemptionPriority`.

Reply:
It can be, while by doing so it also introduces complexity on restoring the priority after waking up. For instance, if a pin is not attached an interrupt in user application while this pin is specified as wakeup source, we cannot call `HAL_Interrupts_Attach()` to restore the priority for this pin.